Emby.Server.Implementations/Updates/InstallationManager.cs

```diff
@@ -529,20 +529,18 @@ private async Task InstallPackageInternal(PackageVersionInfo package, bool isPlu
 
         private async Task PerformPackageInstallation(IProgress<double> progress, string target, PackageVersionInfo package, CancellationToken cancellationToken)
         {
-            // Target based on if it is an archive or single assembly
             var extension = Path.GetExtension(package.targetFilename);
             var isArchive = string.Equals(extension, ".zip", StringComparison.OrdinalIgnoreCase);
 
+            if (!isArchive)
+            {
+                _logger.LogError("Only zip packages are supported. {Filename} is not a zip archive.", package.targetFilename);
+                return;
+            }
+
             if (target == null)
             {
-                if (isArchive)
-                {
-                    target = Path.Combine(_appPaths.PluginsPath, Path.GetFileNameWithoutExtension(package.targetFilename));
-                }
-                else
-                {
-                    target = Path.Combine(_appPaths.PluginsPath, package.targetFilename);
-                }
+                target = Path.Combine(_appPaths.PluginsPath, Path.GetFileNameWithoutExtension(package.targetFilename));
```

Comment:
I don't like the idea of zip file name controlling the directory... we cannot use versions in their names in that case.
Can we make it so there's some metadata inside that is parsed?

Reply:
Sure, however we decide to do this, I can just modify the build infra to generate that sort of information. We just need to add some sort of parsing logic, including e.g. a sha256hash for @EraYaN 's point.

However this is *definitely* getting into a post-10.2.0 territory here. The current setup "works" right now, and if we're changing this much we're already half-way towards doing a proper "repo" setup. I'd rather take our time on this and target it for next release.

Reply:
Oh, and this is just how it was before - it didn't deal with version naming in the .DLL at all - that all comes from the `manifest.json` that the server downloads. In fact, versioning the files themselves will probably break this functionality entirely. Best leave it as-is.

Reply:
The folder name should be the plugin ID from the manifest. And lets keep that human readable like Kodi. `plugin.<catergory>.trakt` for example.

The dll's will have The correct version attribute.

Reply:
The ID is currently the full repo name, but can use a different field. To see what I've got, see jellyfin/jellyfin-plugin-anime#25

Reply:
I'm not saying "fix everything now", rather just expressing my thoughts :)

Reply:
No problem, just getting my thoughts out somewhere :)

Reply:
Would `featureId` be a good field?

Reply:
I think it would be helpful - let's get a forum post going on this (I haven't created one yet, but if someone else can that would be appreciated) and we can get a sensible manifest format worked out!

Reply:
The forum thread would be this one: https://forum.jellyfin.org/t/plugin-distribution/50